pkg/datagen/datagen.go

```diff
@@ -223,13 +234,13 @@ func (c *Cyclic) EdgeChannel() chan *Doc {
 	return c.E
 }
 
-func NewCyclicGraph(n int64) GraphGenerator {
+func NewCyclicGraph(n uint64) GraphGenerator {
```

Comment:
We should change the constructor functions and the whole setup as in https://github.com/arangodb/feed/blob/graphwriter/pkg/graphgen/graphgen.go
The reason is, I need to be able to create a graph, but then only use the vertices (for the vertex collection) or the edges (for the edge collection). But if we start both go routines right away, one of them is leaked. So we should only create the channel and launch the go routine, when somebody askes (via the interface) for the vertex channel (or edge channel respectively).

Reply:
This is for all graph constructions.

Reply:
This is not always as easy as for cycles. Already for trees, we perform some computations to obtain the next vertex and (via the same computations) the net edge, so both are created in the same go routine. We could supply the `Make...Generator` functions with flag parameters saying whether to produce vertices and/or edges. I am not sure, but for constructions like that of lexicographical product, this may become a mess. The reason is that we need both vertices and edges from the graphs to be combined.

Reply:
I would be fine with a solution where we tell the Make...Generator functions with flags what to generate or not, but I need the functionality that I can only create vertices or only create edges. I think it is not a problem to produce a channel too many, but we should not create an unneeded go routine which fills the channel, which would then "leak".